drawBot/drawBotDrawingTools.py

```diff
@@ -1684,19 +1761,19 @@ def fontNamedInstance(self, name, fontNameOrPath=None):
         self._dummyContext._state.text.fontNamedInstance(name, fontNameOrPath)
         self._addInstruction("fontNamedInstance", name, fontNameOrPath)
 
-    def listNamedInstances(self, fontNameOrPath=None):
+    def listNamedInstances(self, fontNameOrPath=None) -> dict[str, dict]:
         return self._dummyContext._state.text.listNamedInstances(fontNameOrPath)
 
     listNamedInstances.__doc__ = FormattedString.listNamedInstances.__doc__
 
-    def textProperties(self):
+    def textProperties(self) -> dict[str, Any]:
         return self._dummyContext._state.text.textProperties()
 
     textProperties.__doc__ = FormattedString.textProperties.__doc__
 
     # drawing text
 
-    def text(self, txt, position, align=None):
+    def text(self, txt: _formattedStringClass | str, position: Point, align: str | None = None):
```

Comment:
This causes an error when running the code (and mypy also complains about it):
```
NameError: name '_formattedStringClass' is not defined. Did you mean: 'FormattedString'?
```

Reply:
Yes, there's a conflict with the function name. Pyright and mypy give different errors, I'll look for a solution.

Reply:
Both type checkers are having a hard time with the function name and the class name being the same. I've opted for a different solution in 7a3199647ccb6f6ab05c7d817277e641920e021a. For the sake of alignment, we could do the same to `BezierPath` and `ImageObject`.

Reply:
I wonder if we should get rid of those "method aliases to class objects", and populate the namespace with the proper classes instead. I never liked the fact that "FormattedString" is a method on the context instead of the intended class.

Reply:
I agree to move BezierPath and ImageObject to proper class objects, but out of scope of this PR

Reply:
MyPy does not complain about this, so for now I'd wait for a new PR.